docs/en/week14/week14-1.md

```diff
@@ -0,0 +1,185 @@
+<img src="https://i.imgur.com/9k4Frq1.png" style="zoom:100%;"/>
+
+<img src="https://i.imgur.com/1k5oVI3.png" style="zoom:100%;"/>
+
+In this case, the energy is a sum of three terms represented by blue squares which are neural networks that produce feature vectors for the input variables. In the case of speech recognition X can be thought of as a speech signal and the squares implement the grammatical constraints and Y represent the generated output labels.
+<!-- RISHABH --------->
+
+<!--- done by Muge --->
+
+## Efficient Inference for Energy-Based Factor Graphs
+
+A Tutorial on Energy-Based Learning (Yann LeCun, Sumit Chopra, Raia Hadsell, Marc’Aurelio Ranzato, and Fu Jie Huang 2006):
+
+Learning and inference with Energy-Based Models involves a minimization of the energy over the set of answers $\mathcal{Y}$ and latent variables $\mathcal{Z}$. When the cardinality of $\mathcal{Y}\times \mathcal{Z}$ is large, this minimization can become intractable. One approach to the problem is to exploit the structure of the energy function in order to perform the minimization efficiently. One case where the structure can be exploited occurs when the energy can be expressed as a sum of individual functions (called factors) that each depend on different subsets of the variables in Y and Z. These dependencies are best expressed in the form of a factor graph. Factor graphs are a general form of graphical models, or belief networks.
+
+<img src="https://i.imgur.com/I56iwGa.png" style="zoom:100%;"/>
+
+
+A simple example of a factor graph is shown in Figure 19 (top). The energy function is the sum of four factors:
+
+$$E(Y, Z, X) = E_a(X, Z_1) + E_b(X, Z_1, Z_2) + E_c(Z_2, Y_1) + E_d(Y_1, Y_2)$$
+
+where $Y = [Y_1, Y_2]$ are the output variables and $Z = [Z_1, Z_2]$ are the latent variables. Each factor can be seen as representing soft constraints between the values of its input variables. The inference problem consists in finding:
+
+$$(\bar{Y}, \bar{Z})=\operatorname{argmin}_{y \in \mathcal{Y}, z \in \mathcal{Z}}\left(E_{a}\left(X, z_{1}\right)+E_{b}\left(X, z_{1}, z_{2}\right)+E_{c}\left(z_{2}, y_{1}\right)+E_{d}\left(y_{1}, y_{2}\right)\right)$$
+
+Let’s assume that $Z_1$, $Z_2$, and $Y_1$ are discrete binary variables, and $Y_2$ is a ternary variable. The cardinality of the domain of $X$ is immaterial since X is always observed. The number of possible configurations of $Z$ and $Y$ given X is $2 \times 2 \times 2 \times 3 = 24$. A naive minimization algorithm through exhaustive search would evaluate the entire energy function 24 times (96 single factor evaluations). 
+
+However, we notice that for a given $X$, $E_a$ only has two possible input configurations: $Z_1 = 0$ and $Z_1 = 1$. Similarly, $E_b$ and $E_c$ only have 4 possible input configurations, and $E_d$ has 6. Hence,
+there is no need for more than 2 + 4 + 4 + 6 = 16 single factor evaluations.
```

Comment:
`$` missing for the math.

Reply:
Hi Alf, 
Could you please refer to where the $ is missing for the math? I didn't find it. Thank you!

Reply:
> `$` missing for the math.

Hi Alf,
Could you please refer to where the $ is missing for the math? I didn't find it. Thank you!

Reply:
```suggestion
there is no need for more than $2 + 4 + 4 + 6 = 16$ single factor evaluations.
```